File: src/main/kotlin/slot.java
package domain;
import java.time.Instant;
import java.util.Objects;
public class slot {
    private final long id;
    private final long containerId;
    private String code;
    private boolean occupied;
    private final Instant createdAt;

    public slot(long id, long containerId, String code) {
        this.id = id;
        this.containerId = containerId;
        this.code = code;
        this.occupied = false;
        this.createdAt = Instant.now();
    }

    public long getId() {
        return id;
    }

    public long getContainerId() {
        return containerId;
    }

    public String getCode() {
        return code;
    }

    public boolean isOccupied() {
        return occupied;
    }

    public void setOccupied(boolean occupied) {
        this.occupied = occupied;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof slot)) return false;
        slot s = (slot) o;
        return id == s.id;
    }

    @Override
    public int hashCode() {
        return Long.hashCode(id);
    }
}

File: src/main/kotlin/Domain/Slot.kt
package Domain

import java.time.Instant

class Slot(
    val id: Long,
    val containerId: Long,
    code: String
) {

    var code: String = code
        set(value) {
            require(value.isNotBlank()) { "Ошибка: code пустой" }
            require(value.length <= 8) { "Ошибка: code слишком длинный" }
            field = value
        }

    var occupied: Boolean = false
    val createdAt: Instant = Instant.now()

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Slot) return false
        return id == other.id
    }

    override fun hashCode(): Int = id.hashCode()
}